src/new/wallet.js

```diff
@@ -159,9 +168,14 @@ class HathorWallet extends EventEmitter {
         console.error('Error loading wallet', {error});
       })
     } else {
-      // CONNECTING or CLOSED?
       this.serverInfo = null;
-      this.setState(HathorWallet.CONNECTING);
+      if (this.firstConnection) {
+        // If firstConnection = true, then a stop() was executed
```

Comment:
Why? What if the connection fails for a while, then it reconnects, and finally we call a stop()?

Reply:
When we call a stop we set `firstConnection = true`. That's why I created this `if`. Before we were changing to 'CONNECTING', then to 'CLOSED' and depending on a race condition it could change to 'CLOSED' first and then 'CONNECTING', which would be wrong because it should be only 'CLOSED'.

Reply:
I still don't understand why we use the `firstConnection` to differentiate both situations. Isn't it better to let the `.stop()` mark a flag (e.g. `stopRequested`) and check it here?

Reply:
Makes sense, I created `walletStopped` flag to check in this `if`